SE Development Models (#32)
Vaibhavpratapsingh22/Courses: 4-software-engineering/README.md

# Software Engineering

Software Engineering is the field of computer science that deals with the construction of large or complex software system. The software development of this magnitude is usually done by a group of software professionals rather than the single professional. The differnce between programming and software engineering is that programming is primiarly activity while software engineering is essentially a team activity.

## Chapter 1 - Introduction

It is about the basics of software engineering. We will start with how a program is different from software. Then we will head towards the software crisis. In the software crisis, we will learn its causes, characteristics. At last, we will learn about the elements of software engineering.

## Chapter 2 -  Software Development Process Models

> In this chapter, we learned about Software Development Process Models and different phases through which we build software. After this, we had a detailed explanation of the waterfall model with the help of a diagram . Finally, we closed with waterfall model applications, advantages, and disadvantages.

Reviewer: ```suggestion
It is about Software Development Process Models and different phases through which we build software. After this, the waterfall model is explained in detail with the help of a diagram . At last, we will learn about the waterfall model applications, advantages, and disadvantages.
```

# Software Engineering

Software Engineering is the field of computer science that deals with the construction of large or complex software system. The software development of this magnitude is usually done by a group of software professionals rather than the single professional. The differnce between programming and software engineering is that programming is primiarly activity while software engineering is essentially a team activity.

You will gain not only knowledge and practical experience of the latest technologies, but also a grounding in the underlying principles of the subject. It is this combination of skills that will us to keep pace with this fast-evolving subject!

## Chapter 1 - Introduction

In this chapter we will look at the basics of software engineering like :

1. Definition of Software
   * Program versus Software
   * Software Crisis
2. Software Characteristics
3. Definition of Software Engineering
4. Elements of Software Engineering

> 

Reviewer: There is no chapter 2 discussion here.

Reviewer: This is still missing. I have made a suggestion on another branch for how this abstract could be written. You can write something similar for chapter 2 as well in the SE README file.
